enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,915 @@
+4. Provide a way for index authors to reason about their indexes via a visual representation of the index, without having to curl a grpc api that in turn queries a sqllite database underneath, etc.
+5. Provide a way for tools like `opm`, `operator-registry` etc to consume the representation of an index instead of needing a sqlite database as an input whenever they need information about the index to perform a task. 
+6. Allow for reproducibility of an index using just the yaml/json representation of the index. 
+
+## Non-goals 
+
+1. Enumerate/implement tooling to allow different operations to be performed on the package representations. Numerous tooling(eg those that allow operation on files) that already exists can be leveraged to perform various operations on package representations.    
+
+## User Stories 
+
+### Story 1
+
+As a user with pull permission from the namespace an index is hosted in/as a component of operator-lifecycle-manager, I can query an index for a json/yaml representation of the packages of an index.
+The representation of individual packages will allow individual package owners to reason about/make changes to their individual packages in isolation.  
+
+`opm index inspect --index=docker.io/my-namespace/community-operators --output=json`
+
+creates a new folder `community-operators` and downloads the package representations for the packages that are in the index
+
+```bash
+$ cd community-operators
+$ tree
+.
+├── etcd.json
+├── amqstreams.json
+├── image_sha256.json
+
+$ cat etcd.json
+[
+    {
+        "schema": "package.v1",
+        "name": "etcd",
+        "defaultChannel": "singlenamespace-alpha",
+        "icon": {
+            "base64data":"iVBORw0KGgoAAAANSUhEUgAAA.....",
+            "mediatype":"image/png"
+        },
+        "channels": ["alpha", "singlenamespace-alpha", "clusterwide-alpha"],
+        "description": "A message about etcd operator, a description of channels"
+    },
+    {
+        "schema": "bundle.v1", 
+        "name": "etcdoperator-community.v0.6.1",
+        "package": "etcd",
+        "image": "quay.io/operatorhubio/etcd:v0.6.1",
+        "version": "0.6.1",
+        "properties":[
+            {
+                "name": "olm.package",
+                "values": {
+                    "packageName": "etcd",
+                    "version": "0.6.1"
+                }
+            },
+            {
+                "name":"olm.gvk",
+                "values":{
+                    "group": "etcd.database.coreos.com", 
+                    "kind": "EtcdCluster", 
+                    "version": "v1beta2"
+                }
+            }
+        ],
+        "channels": [
+            {
+                "name": "alpha",
+                "upgradeInfo": {}
+            }
+        ],
+        "relatedImages": [
+            "quay.io/coreos/etcd-operator@sha256:bd944a211eaf8f31da5e6d69e8541e7cada8f16a9f7a5a570b22478997819943"
```

Comment:
why do we need this in the DB?  if we get rid of it it removes one obstacle to maintaining this file for teams...

Reply:
sorry i guess this is supposed to be a ref to the bundle image sha?  which obviously we need.  I read it as other related images (like a CSV would have to point to operand images).

but it definitely remains problematic for operator-pipeline flows as noted in my overall comment.

Reply:
Oh, this is a good point. *All* of the related images (operand images, bundle images, etc) *needed* to be in the database otherwise disconnected mirroring would take forever (we'd have to unpack each bundle one at a time to generate the list). So if we're trying to abstract the SHAs away, we would either need some way to associate them with the disconnected indexes. I think we need a way for this content to be generated or replaced at index build time.